vite-plugin-ssr/node/renderPage.ts

```diff
@@ -72,12 +72,18 @@ export { throwPrerenderError }
 type PageFiles = PromiseType<ReturnType<typeof loadPageFiles>>
 type GlobalContext = PromiseType<ReturnType<typeof getGlobalContext>>
 
-async function renderPage<PageContextAdded extends {}, PageContextInit extends { url: string }>(
-  pageContextInit: PageContextInit,
-): Promise<
-  PageContextInit &
-    (({ httpResponse: HttpResponse } & PageContextAdded) | ({ httpResponse: null } & Partial<PageContextAdded>))
-> {
+interface RenderPage<PageContextAdded extends {}> {
+  <PageContextInit extends { url: string }>(
+    pageContextInit: PageContextInit,
+  ): Promise<
+    PageContextInit &
+      (({ httpResponse: HttpResponse } & PageContextAdded) | ({ httpResponse: null } & Partial<PageContextAdded>))
+  >
+}
+
+const renderPage: RenderPage<{}> = async function renderPage (
+  pageContextInit,
+) {
```

Comment:
`PageContextAdded` needs to be specified, but `PageContextInit` can always be infered. I tested it that way:

```ts
const renderPage = createPageRenderer<RenderPageCustomFields>({
  isProduction: true,
});

...
const pageContextInit = { url: req.url, body: req.body, cookies: req.cookies };
const pageContext = await renderPage(pageContextInit);
// pageContext typing contains Partial<RenderPageCustomFields> and type infered from pageContextInit
```

Reply:
👍 That's a neat idea and neat trick with `interface`.